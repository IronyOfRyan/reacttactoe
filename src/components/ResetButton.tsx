import { Button } from "../view/styles";
import { calculateWinner } from "../util/gameAI";

export interface ButtonProps {
  state: {
    readonly squares: (string | null)[];
  };
  handleReset: () => void;
}

const ResetButton = ({ state, handleReset }:ButtonProps):JSX.Element => {
  let winner = calculateWinner(state.squares);
  // Hides reset button when the game's over
  if (winner || !state.squares.includes(null)) {
    return <Button gameover={true}/>;
  } else {
    return (
      <Button 
        gameover={false} 
        onClick={() => handleReset()}
        onTouchEnd={() => handleReset()} 
      >
        RESET
      </Button>
    );
  }
};

export default ResetButton;