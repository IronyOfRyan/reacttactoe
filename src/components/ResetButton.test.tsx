import ReactDOM from "react-dom";
// import { render, screen } from '@testing-library/react';
import ResetButton from "./ResetButton";

test("renders without crashing", () => {
  const div = document.createElement("div");
  const state = {squares: []}  
  const handleReset = (() => null);
  ReactDOM.render(<ResetButton state={state} handleReset={handleReset}/>, div);
});
