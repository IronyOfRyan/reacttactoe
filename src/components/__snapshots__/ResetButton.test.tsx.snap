// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check styled-component styling 1`] = `
.c0 {
  background: #cbb4d4;
  border: 0.5px solid #a044ff;
  box-shadow: 0 0 0 2px #cbb4d4,2px 1px 2px 3px rgba(10,10,0,0.5);
  color: #a044ff;
  font-family: Fascinate,sans-serif;
  font-weight: bold;
  font-size: 2em;
  height: 2.5em;
  line-height: 2em;
  padding: 1px;
  text-align: center;
  width: 8.5em;
  visibility: hidden;
}

.c0:hover {
  background: #a044ff;
  border: 1px dashed #a044ff;
  box-shadow: 0 0 0 2px #a044ff,2px 1px 2px 3px rgba(10,10,0,0.5);
  text-shadow: -1px #cbb4d4;
  color: #cbb4d4;
}

.c0:focus {
  outline: none;
}

@media (min-width:1024px) {
  .c0 {
    font-size: 1.2em;
    height: 1.5em;
    line-height: 1em;
  }
}

<button
  class="c0"
/>
`;
